### ISSUE TITLE
Scriptability of Oni

### ISSUE THREAD

**Issue by bert88sta** [extr0py/oni#610]
How scriptable is Oni right now? I know that we have an existing plugin infrastructure, but if I miss the whole "Just add it to your .vimrc" mentality for small things

**Comment by extr0py** [comment]
Great question. This is something I've been thinking about.

One thing I'd like to do is treat the `Config.js` as a mini-plugin, and have an `activate` method for it - so you can script against the `Oni` object.

You can also open the developer tools and make calls against the `Oni` object (which is the same thing the plugins get), but it's not super interesting - it'll be moreso once there are things like keybindings hooked up... 

Is there a particular sort of customization you were thinking about, that wasn't easy to do without rebuilding Oni?

**Comment by bert88sta** [comment]
Maybe a dumb example, but hypothetically I might want to change the
background color of the whole status bar based on mode. I don't actually,
but it seems like something I *should* be able to do. I don't know that
it's going to be feasible in JS and electron, but basically any state that
isn't critical to the editor's correct functionality should be available to
the user

**Comment by extr0py** [comment]
Not a dumb example at all, this is great, and is exactly what I'd like to support.

The way I'm thinking about this would be, in `Config.js`, we'd expose an activate method:
```
const activate = (Oni) => {
     
     Oni.on("mode-changed", (newMode) => {
          const newColor = newMode === "insert" ? "black" : "white"
           Oni.configuration.setValue("oni.colors", {
                 "statusbar.background": newColor
           })
     })
}

module.exports = {
   activate,

  //add custom config here, such as
  //"oni.useDefaultConfig": true,
  "editor.fontSize": "12px",
  "editor.fontFamily": "Fira Code",
  "oni.bookmarks": ["C:/Toronto2"],
  "oni.loadInitVim": true,
  "statusbar.enabled": true,
  "editor.scrollBar.visible": true,
  "editor.cursorLine": true,
  "editor.cursorLineOpacity": 0.1,
...
```

Something like that... It's a bit awkward having the function in the same space as the configuration, but I'd like to use 'activate' and 'deactivate' as reserved capabilities here. 

The setting of color properties ties into some of the thoughts @cyansprite had in #604 

Ideally, it should be really lightweight... in that I can save and see changes immediately, as well as be very straightforward - it should have premier autocompletion / quickinfo / etc.

**Comment by extr0py** [comment]
Here's another scriptability example I was thinking of the other day - I'd like to make an easy way to take screenshots of the window (or, later, even record/pause/stop gifs/videos):

```
Oni.input.bind("<C-Space>", () => Oni.workspace.takeScreenshot())
```

**Comment by bryphe** [comment]
FYI, the screenshot example works now in 0.2.12. 

I've started documenting the API surface here:
https://github.com/bryphe/oni/wiki/API

And some examples of how to use it in the config:
https://github.com/bryphe/oni/wiki/Configuration#examples

**Comment by TalAmuyal** [comment]
@bryphe AFAIK, this was implemented.
Am I correct? (if so, let's close this issue :D)

**Comment by badosu** [comment]
@TalAmuyal I agree, seems like basic scriptability is available, only thing missing is documentation on the API functions available.
